js/tools/tooloption.js

```diff
@@ -0,0 +1,153 @@
+            }
+        });
+
+        colorSelectTextbox.addEventListener("keyup", e => {
+            if (e.keyCode === 13) {
+                colorSelectButton.click();
+            }
+        })
+
+        optionsElement.appendChild(optionsTitle);
+        optionsElement.appendChild(colorList);
+        optionsElement.appendChild(colorSelectTextbox);
+        optionsElement.appendChild(colorSelectButton);
+
+        return optionsElement;
+    }
+
+    // Creates a colored square div for selecting fill color and attaches a click event listener
+    createColorSelectorElement(color) {
+        var element = document.createElement("div");
+        var lowercasedColor = color.toLowerCase();
+    
+        element.style.display = "inline-block"
+        element.style.backgroundColor = lowercasedColor;
+        element.style.width = "20px";
+        element.style.height = "20px";
+        element.style.margin = "5px";
+        element.style.border = "solid black 1px";
+    
+        element.addEventListener("click", e => {
+            console.log(lowercasedColor);
+            this.fill.color = lowercasedColor;
+            document.getElementById("options-title").textContent = `Fill Color: ${this.fill.color}`;
+        });    
+    
+        return element;
+    }
+
+    strokeOptions() {
+        this.stroke = {}; // Settings about the stroke of a tool are stored in {ToolClass}.options.strokeSettings
+        this.stroke["thickness"] = 3;
+
+        var thicknessSlider = document.createElement("input");
+        thicknessSlider.type = "range";
+        thicknessSlider.min = 1;
+        thicknessSlider.max = 10;
+        thicknessSlider.value = this.stroke["thickness"];
+
+        thicknessSlider.addEventListener("input", e => {
+            this.stroke["thickness"] = e.target.value;
+            optionsTitle.textContent = `Stroke Thickness: ${this.stroke['thickness']}`;
+        })
+
+        var optionsElement = document.createElement("div");
+        var optionsTitle = document.createElement("span");
+        optionsTitle.textContent = `Stroke Thickness: ${this.stroke['thickness']}`;
+
+        optionsElement.appendChild(optionsTitle);
+        optionsElement.appendChild(thicknessSlider);
+
+        return optionsElement;
+    }
+
+    // Go through each option defined in the array and call their associated methods stored in optionMapping
+    // to get the DOM elements to display in the customization panel
+    render() {
+        var toolCustomizationPanel = document.getElementById("tool-customization");
+
+        while (toolCustomizationPanel.hasChildNodes()) {
+            toolCustomizationPanel.removeChild(toolCustomizationPanel.lastChild);
+        }
+
+        this.optionDOMElements.forEach(optionDOM => {
+            toolCustomizationPanel.appendChild(optionDOM);
+        });
+    }
+}
+
+// Checks if given string is a valid CSS color
+function isValidCSSColor(string) {
+    var s = new Option().style;
```

Comment:
if this is the method of choice for verifying css string is it possible to use a class other than `Option`? It's pretty confusing here